src/main/java/htsjdk/tribble/readers/AsciiLineReader.java

```diff
@@ -40,11 +40,19 @@
     private static final byte LINEFEED = (byte) ('\n' & 0xff);
     private static final byte CARRIAGE_RETURN = (byte) ('\r' & 0xff);
 
-    private final PositionalBufferedStream is;
+    private PositionalBufferedStream is;
     private char[] lineBuffer;
     private int lineTerminatorLength = -1;
 
+    protected AsciiLineReader() {};
+
+    /**
+     * TODO: This class implements LocationAware, which requires preservation of virtual file pointers on BGZF inputs.
+     * However, if the inputStream wrapped by this class is a BlockCompressedInputStream, it violates that contract by
+     * wrapping the stream and returning positional file offsets instead.
+     */
     public AsciiLineReader(final InputStream is){
+        // NOTE: This will wrap the input stream in a PositionalBufferedStream even if its already a PositionalBufferedStream
```

Comment:
Not sure it should be illegal to pass a BCS to AsciiLineReader (my comment that it violates the contract might be an overstatement). Perhaps we should just change the doc to be more explicit about the fact that it provides buffering, which affects the positions returned.